model/service.go

```diff
@@ -25,10 +25,15 @@ import (
 type ServiceDiscovery interface {
 	// Services list declarations of all services and their tags
 	Services() []*Service
+
+	// GetService retrieves a service by name if it exists
+	GetService(name, namespace string) (*Service, bool)
```

Comment:
There is `Services` method that is not supposed to do what you want. This is an optimized method to get a service by a name. The resulting service contains a list of tags.